src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -84,19 +85,34 @@ internal AbsolutePath(string? directory, string fileName, IFileSystem fileSystem
     /// </remarks>
     public static AbsolutePath FromFullPath(string fullPath, IFileSystem? fileSystem = null)
     {
-        var index = fullPath.LastIndexOf(Path.DirectorySeparatorChar);
-        if (index < 0)
-        {
-            // No directory
+        var span = fullPath.AsSpan();
+
+        // path is not rooted
+        var rootLength = GetRootLength(span);
+        if (rootLength == 0)
             return new AbsolutePath(null, fullPath, fileSystem ?? Paths.FileSystem.Shared);
+
+        // path is only the root directory
+        if (span.Length == rootLength)
+            return new AbsolutePath(fullPath, "", fileSystem ?? Paths.FileSystem.Shared);
+
+        var slice = span[rootLength..];
```

Comment:
Use `SliceFast` here to elide bounds checks. 
We know the root here isn't out of bounds because it is either 0 or found; but the JIT isn't smart enough to pick that up.

For the lines below, it seems a bounds check is still emitted as JIT can't tell that slice should be length > 1 [this should be guaranteed by above checks].

So for `slice[^1]` use the following method to index without a bounds check [taken from `CommunityToolkit.HighPerformance`]

```csharp
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

/// <summary>
/// Returns a reference to an element at a specified index within a given <see cref="Span{T}"/>, with no bounds checks.
/// </summary>
/// <typeparam name="T">The type of elements in the input <see cref="Span{T}"/> instance.</typeparam>
/// <param name="span">The input <see cref="Span{T}"/> instance.</param>
/// <param name="i">The index of the element to retrieve within <paramref name="span"/>.</param>
/// <returns>A reference to the element within <paramref name="span"/> at the index specified by <paramref name="i"/>.</returns>
/// <remarks>This method doesn't do any bounds checks, therefore it is responsibility of the caller to ensure the <paramref name="i"/> parameter is valid.</remarks>
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static ref T DangerousGetReferenceAt<T>(this Span<T> span, int i)
{
    ref T r0 = ref MemoryMarshal.GetReference(span);
    ref T ri = ref Unsafe.Add(ref r0, (nint)(uint)i);

    return ref ri;
}
```